core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -177,10 +179,16 @@ private static BlockInStream createGrpcBlockInStream(FileSystemContext context,
     long chunkSize = context.getClusterConf().getBytes(
         PropertyKey.USER_STREAMING_READER_CHUNK_SIZE_BYTES);
     readRequestBuilder.setChunkSize(chunkSize);
-    DataReader.Factory factory =
-        new GrpcDataReader.Factory(context, address, readRequestBuilder.build());
-    return new BlockInStream(factory, address, blockSource, readRequestPartial.getBlockId(),
-        blockSize);
+    DataReader.Factory factory;
+    if (context.getClusterConf().getBoolean(PropertyKey.FUSE_SHARED_CACHING_READER_ENABLED)
+        && (blockSize > (chunkSize * 4))) {
```

Comment:
Is this (`(blockSize > (chunkSize * 4)`) some heuristic? What is the thinking behind this?

Reply:
yes this is a heuristic motivated by tests done by MSFT and verified & contributed by them too.
essentially, for concurrent reads for the same large file through FUSE, the reads likely ended up at different FileInputStream, and introduce frequent blockInStream construction & destruction, causing performance down by 100x.

this fix can effectively mitigate this issue. Note that, this heuristic is always guarded by the flag